src/Libraries/DSCPython/DCPythonCodeCompletionProviderCore.cs

```diff
@@ -0,0 +1,394 @@
+﻿using Autodesk.DesignScript.Interfaces;
+using Dynamo.Logging;
+using Dynamo.PythonServices;
+using Python.Runtime;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace DSCPython
+{
+    internal class DSCPythonCodeCompletionProviderCore : PythonCodeCompletionProviderCommon, ILogSource, IDisposable
+    {
+        #region Private members
+
+        /// <summary>
+        /// The engine used for autocompletion.  This essentially keeps
+        /// track of the state of the editor, allowing access to variable types and
+        /// imported symbols.
+        /// </summary>
+        private PyScope scope;
+
+        private string uniquePythonScopeName = "uniqueScope";
+
+        /// <summary>
+        /// The scope used by the engine.  This is where all the loaded symbols
+        /// are stored.  It's essentially an environment dictionary.
+        /// </summary>
+        private PyScope Scope
+        {
+            get { return scope; }
+            set { scope = (PyScope)value; }
+        }
+
+        private PyScope CreateUniquePythonScope()
+        {
+            PyScopeManager.Global.TryGet(uniquePythonScopeName, out PyScope Scope);
+
+            if (Scope == null)
+            {
+                Scope = Py.CreateScope(uniquePythonScopeName);
+                Scope.Exec(@"
+import clr
+import sys
+clr.setPreload(True)
```

Comment:
I'm unclear if this is required here:
https://github.com/DynamoDS/Dynamo/pull/10950 for more info.

Reply:
I created a static global scope which will do the Preload 
I guess the only duplication now is between the Evaluator's preloading and the Autocomplete's preloading